src/components/CreateGraph/CreateGraph.jsx

```diff
@@ -30,131 +27,181 @@ const CreateGraph = () => {
+        if (!isPlacingNode) return;
+        const boundingRect = event.target.getBoundingClientRect();
 
     const mouseX = event.clientX - boundingRect.left;
     const mouseY = event.clientY - boundingRect.top;
 
-    const newNode = { x: mouseX, y: mouseY };
+        const newNode = { x: mouseX, y: mouseY, isSelected: false, key: nodes.length + 1 };
 
     const nodeExists = nodes.some(
       (node) => node.x === newNode.x && node.y === newNode.y
     );
 
-    if (nodeExists) {
-      message.error(texts.texts.CreateGraphErrors.nodeExists);
-      return;
-    }
-
-    setNodes([...nodes, newNode]);
-    setIsPlacingNode(false);
-  };
-
-  const addEdge = (sourceNode, targetNode) => {
-    const newEdge = { source: sourceNode, target: targetNode };
-
-    const edgeExists = edges.some(
-      (edge) =>
-        (edge.source === newEdge.source && edge.target === newEdge.target) ||
-        (edge.source === newEdge.target && edge.target === newEdge.source)
-    );
-    if (edgeExists) {
-      message.error('The edge already exists in the graph.');
-      return;
-    }
-
-    setEdges([...edges, newEdge]);
-  };
-
-  const deleteNode = () => {
-    if (nodes.length === 0) {
-      message.error(texts.texts.CreateGraphErrors.noNodes);
-      return;
-    }
-    const updatedNodes = [...nodes];
-    updatedNodes.pop();
-    setNodes(updatedNodes);
-  };
+        if (nodeExists) {
+            message.error(texts.NODE_EXISTS);
+            return;
+        }
 
-  const deleteEdge = () => {
-    if (edges.length === 0) {
-      message.error(texts.texts.CreateGraphErrors.noEdges);
-      return;
+        setNodes([...nodes, newNode]);
+        setIsPlacingNode(false);
     }
-    const updatedEdges = [...edges];
-    updatedEdges.pop();
-    setEdges(updatedEdges);
-  };
-
-  const clearAll = () => {
-    setNodes([]);
-    setEdges([]);
-  };
-
-  const buttons = [
-    {
-      text: 'Add Node',
-      icon: <PlusCircleFilled />,
-      onClick: handleStartPlacingNode,
-    },
-    { text: 'Delete Node', icon: <MinusCircleFilled />, onClick: deleteNode },
-    { text: 'Delete Edge', icon: <MinusCircleFilled />, onClick: deleteEdge },
-    { text: 'Clear All', icon: <DeleteFilled />, onClick: clearAll },
-  ];
-
-  return (
-    <>
-      <Flex gap="L" align="center" vertical>
-        <Flex gap="small" wrap="wrap">
-          {buttons.map((button, index) => (
-            <Button
-              key={index}
-              type="primary"
-              size="large"
-              icon={button.icon}
-              onClick={button.onClick}
+    const addEdge = (sourceNode, targetNode) => {
+
+        const sourceNodeId = sourceNode.key;
+        const targetNodeId = targetNode.key;
+
+        const edgeExists = edges.some(
+            (edge) =>
+                (edge.source.key === sourceNodeId && edge.target.key === targetNodeId) ||
+                (edge.source.key === targetNodeId && edge.target.key === sourceNodeId)
+        );
+
```

Comment:
Please use destruction to avoid code duplication: 

```js
const edgeExists = edges.some(({ source, target }) => {
    const { key: sourceKey } = source;
    const { key: targetKey } = target;
    return (
        (sourceKey === sourceNodeId && targetKey === targetNodeId) ||
        (sourceKey === targetNodeId && targetKey === sourceNodeId)
    );
});
```